internal/app/vodtester/continuous_vod_tester.go

```diff
@@ -0,0 +1,177 @@
+package vodtester
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/PagerDuty/go-pagerduty"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/stream-tester/internal/testers"
+	"github.com/livepeer/stream-tester/messenger"
+)
+
+type (
+	// IContinuousVodTester ...
+	IContinuousVodTester interface {
+		// Start start test. Blocks until error.
+		Start(fileName string, testDuration, pauseDuration, pauseBetweenTests time.Duration) error
+		Cancel()
+		Done() <-chan struct{}
+	}
+
+	ContinuousVodTesterOptions struct {
+		PagerDutyIntegrationKey string
+		PagerDutyComponent      string
+		PagerDutyLowUrgency     bool
+		VodTesterOptions
+	}
+
+	continuousVodTester struct {
+		ctx                     context.Context
+		cancel                  context.CancelFunc
+		host                    string // API host being tested
+		pagerDutyIntegrationKey string
+		pagerDutyComponent      string
+		pagerDutyLowUrgency     bool
+		vtOpts                  VodTesterOptions
+	}
+)
+
+// NewContinuousVodTester returns new object
+func NewContinuousVodTester(gctx context.Context, opts ContinuousVodTesterOptions) IContinuousVodTester {
+	ctx, cancel := context.WithCancel(gctx)
+	server := opts.API.GetServer()
+	u, _ := url.Parse(server)
+	cvt := &continuousVodTester{
+		ctx:                     ctx,
+		cancel:                  cancel,
+		host:                    u.Host,
+		pagerDutyIntegrationKey: opts.PagerDutyIntegrationKey,
+		pagerDutyComponent:      opts.PagerDutyComponent,
+		pagerDutyLowUrgency:     opts.PagerDutyLowUrgency,
+		vtOpts:                  opts.VodTesterOptions,
+	}
+	return cvt
+}
+
+func (cvt *continuousVodTester) Start(fileName string, testDuration, pauseDuration, pauseBetweenTests time.Duration) error {
+	messenger.SendMessage(fmt.Sprintf("Starting continuous vod test of %s", cvt.host))
+	try := 0
+	notRtmpTry := 0
+	maxTestDuration := 4*testDuration + pauseDuration + 15*time.Minute
+	for {
```

Comment:
WDYT of enforcing a test cadence here, just so we don't do tests as quickly as possible on the VOD API? Say soon we might optimize the transcode/prepare tasks and then this tests would start flying really fast.

So here, we can actually use this `testDuration` which already exists and is passed around by the `record-tester` command to force a minimum delay between the tests. You can enforce it easily here with a ticker!
```suggestion
func (cvt *continuousVodTester) Start(fileName string, testDuration, pauseDuration, pauseBetweenTests time.Duration) error {
	messenger.SendMessage(fmt.Sprintf("Starting continuous vod test of %s", cvt.host))
	try := 0
	notRtmpTry := 0
	maxTestDuration := 4*testDuration + pauseDuration + 15*time.Minute
	ticker := time.NewTicker(testDuration)
	defer ticker.Stop()
	for range ticker.C {
```
And so the loop will only be "triggered" every `testDuration` 😌 

WDYT?